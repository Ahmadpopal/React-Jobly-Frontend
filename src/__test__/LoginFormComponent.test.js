import React from 'react';
import UserContext from '../Context';
import { render } from '@testing-library/react';
import { MemoryRouter} from 'react-router-dom';
import Login from '../LogInComponent'


describe('LOGIN TESTS', () => {

    // SMOKE TEST 
    it('Login Component should render without crushing', ()=>{
        let currentUser = null;
        render(
            <UserContext.Provider value={{ currentUser }} >
                <MemoryRouter>
                    <Login />
                </MemoryRouter>
            </UserContext.Provider>
        )
    }) 

    // SNAPSHOT TEST 
    it('', () => {
        let currentUser = null;
        const { asFragment } = render(
            <UserContext.Provider value={{ customElements}} >
            <MemoryRouter>
                <Login />
            </MemoryRouter>
        </UserContext.Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    })


    it('shows a login form', () => {
		let currentUser = null;
		const { getByText } = render(
			<UserContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Login />
				</MemoryRouter>
			</UserContext.Provider>
		);
		expect(getByText('Username')).toBeInTheDocument();
	});

});