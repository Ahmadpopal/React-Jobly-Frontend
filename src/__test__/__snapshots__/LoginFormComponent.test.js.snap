// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOGIN TESTS  1`] = `
<DocumentFragment>
  <div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="Login-div-title col"
        >
          <h3
            class="Login-h3"
          >
            Log In
          </h3>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <form
            class="Login-form"
          >
            <div
              class="form-group"
            >
              <label
                class="Login-label"
                for="username"
              >
                Username
              </label>
              <input
                class="Login-Input form-control"
                id="username"
                name="username"
                placeholder="Username Here"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="Login-label"
                for="password"
              >
                password
              </label>
              <input
                class="Login-Input form-control"
                id="password"
                name="password"
                placeholder="Password Here"
                type="password"
                value=""
              />
            </div>
            <button
              class="Login-button btn btn-success"
            >
              Log In
            </button>
          </form>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="Login-signUp-div col"
        >
          <p
            class="Login-signUp-p"
          >
            New User ? 
          </p>
          <a
            class="Login-signUp-link"
            href="/signup"
          >
            Create an account.
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
